r/tests/testthat/test-dataset-write.R

```diff
@@ -458,6 +458,8 @@ test_that("Writing a dataset: CSV format options", {
 test_that("Dataset writing: unsupported features/input validation", {
   skip_if_not_available("parquet")
   expect_error(write_dataset(4), "'dataset' must be a Dataset, ")
+  expect_error(write_dataset(data.frame(x = 1, x = 2, check.names = FALSE)),
+               "Duplicated field names")
```

Comment:
Thanks for adding this test, great to have this looked at here explicitly!

I know you didn't write the original error message, but while we're looking at this, I wonder if we could also look to improve it?  The [tidyverse style guide](https://style.tidyverse.org/error-messages.html) suggests the use of "must" or "can't", so we could perhaps change it to something like "Field names must be unique".